apps/dev-portal-web/domains/docs/components/widgets/AmountDistributionCalculator/index.tsx

```diff
@@ -0,0 +1,217 @@
+import { Col, Row, RowProps } from 'antd'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useIntl } from 'react-intl'
+
+import { split, createRecipientKey } from '@open-condo/billing/utils/paymentSplitter'
+import type { TSplit, TDistributionItem, TSplitOptions } from '@open-condo/billing/utils/paymentSplitter'
+import { Plus, Trash } from '@open-condo/icons'
+import { Alert, Button, Card, Input, MarkdownCodeWrapper, Tabs, Typography } from '@open-condo/ui'
+
+import { DistributionItem } from './DistributionItem'
+import { SplitResult } from './SplitResult'
+
+const CARD_PADDING = 8
+const GUTTER_ROW: RowProps['gutter'] = [16, 16]
+
+function generateDecimalString (len: number): string {
+    return Math.random().toString().substring(2, len) // NOSONAR
```

Comment:
What's the substring used for?

Reply:
Btw, this way you cannot guarantee length...Since if your random is `0.0100000`, then you've got `"0.01"` as string and `"01"` as substring. Even if length is 8.

Reply:
Why not something like this? It will guarantee the length
```suggestion
    return String(Math.floor(Math.random() * Math.pow(10, length))).padStart(length, '0')
```

Reply:
This is a random string that looks like some number for the recipient card. Any string is fit here. There is no need for any guarantees :)

Reply:
You accept length as argument and return any string, that's weird

Reply:
Refactored